File: Demo05_MVP/src/main/java/me/honge/demo05_mvp/ui/base/BaseActivity.java
package me.honge.demo05_mvp.ui.base;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;

import com.socks.library.KLog;

import me.honge.demo05_mvp.App;
import me.honge.demo05_mvp.R;
import me.honge.demo05_mvp.injection.component.ActivityComponent;
import me.honge.demo05_mvp.injection.component.DaggerActivityComponent;
import me.honge.demo05_mvp.injection.module.ActivityModule;
import me.honge.demo05_mvp.ui.main.MainActivity;

/**
 * Created by hong on 16/1/4.
 */
public class BaseActivity extends AppCompatActivity {
    private ActivityComponent mActivityComponent;
    private Toolbar toolbar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        KLog.e();
        super.onCreate(savedInstanceState);
    }

    @Override
    public void setContentView(int layoutResID) {
        super.setContentView(layoutResID);
        initToolBar();
    }

    @Override
    public void setContentView(View view) {
        super.setContentView(view);
        initToolBar();
    }

    @Override
    public void setContentView(View view, ViewGroup.LayoutParams params) {
        super.setContentView(view, params);
        initToolBar();
    }

    private void initToolBar() {
        toolbar = (Toolbar) findViewById(R.id.realToolbar);
        if (toolbar != null) {
            setSupportActionBar(toolbar);

            getSupportActionBar().setHomeButtonEnabled(true);
            getSupportActionBar().setDisplayShowHomeEnabled(true);
            if (this instanceof MainActivity) {
                getSupportActionBar().setDisplayHomeAsUpEnabled(false);
            } else {
                getSupportActionBar().setDisplayHomeAsUpEnabled(true);
            }
        }

    }

    public Toolbar getToolbar() {
        return toolbar;
    }

    public ActivityComponent getComponent() {
        if (mActivityComponent == null) {
            mActivityComponent = DaggerActivityComponent.builder()
                    .activityModule(new ActivityModule(this))
                    .applicationComponent(App.get(this).getComponent())
                    .build();
        }
        return mActivityComponent;
    }
    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                if (this instanceof MainActivity) {

                } else {
                    // finish();
                    onBackPressed();
                }

                break;
        }
        return super.onOptionsItemSelected(item);
    }
}

File: Demo09_PassWord/src/main/kotlin/me/honge/demo09_password/ui/base/BaseActivity.kt
package me.honge.demo09_password.ui.base

import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar
import android.view.MenuItem
import android.view.View
import android.view.ViewGroup

import me.honge.demo09_password.R
import me.honge.demo09_password.ui.main.MainActivity

/**
 * Created by hong on 16/2/24.
 */
open class BaseActivity : AppCompatActivity() {
    var toolbar: Toolbar? = null
        private set

    override fun setContentView(layoutResID: Int) {
        super.setContentView(layoutResID)
        initToolBar()
    }

    override fun setContentView(view: View) {
        super.setContentView(view)
        initToolBar()
    }

    override fun setContentView(view: View, params: ViewGroup.LayoutParams) {
        super.setContentView(view, params)
        initToolBar()
    }

    private fun initToolBar() {
        toolbar = findViewById(R.id.realToolbar) as Toolbar?
        if (toolbar != null) {
            setSupportActionBar(toolbar)

            supportActionBar!!.setHomeButtonEnabled(true)
            supportActionBar!!.setDisplayShowHomeEnabled(true)
            if (this is MainActivity) {
                getSupportActionBar()!!.setDisplayHomeAsUpEnabled(false)
            } else {
                supportActionBar!!.setDisplayHomeAsUpEnabled(true)
            }
        }

    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> if (this is MainActivity) {

            } else {
                // finish();
                onBackPressed()
            }
        }
        return super.onOptionsItemSelected(item)
    }
}
